### DISCUSSION TITLE
add a type to a TemporaryUploadedFile leads to a 500 Error

### DISCUSSION THREAD

**Discussion by michabbb** [livewire/livewire#4658] [1. Help] [answered]
hi there,

i made my first file upload and followed the [documentation](https://laravel-livewire.com/docs/2.x/file-uploads).

so I added a public property in my component class:

```
public $photo;
```

and of course, my IDE tells me, it would be nice give that property a type 😏 
so I added `?TemporaryUploadedFile` but adding the type leads to a 500 error.
so in other words, this is not working:

```
public ?TemporaryUploadedFile $photo = NULL;
```

when I submit my form, `$photo` has become a class of `TemporaryUploadedFile`, everything is working without the type,
but with I get the 500 error - and I don´t see any errors in the laravel.log - so I know it´s because of the type, so I
want to ask:

- is there any way to make this work **with** the type ?
- how in general can I debug a livewire 500 error, because the laravel.log is empty? so if I wouldn´t know it´s because of the type it would be hard to find the cause for the error

thanks for any help/advice!

**Comment by joshhanley** [comment]
@michabbb hmm it's possible that a string gets added to it first. Seems strange you would get a 500 error though without any details with what is going on. Do you have `APP_DEBUG` set to true in your env file?

**Reply by michabbb** [reply]
thanks for your feedback. i don´t really know what has changed, maybe it was a browser issue that I only saw a black screen. after reproducing the issue I finally see a regular error page. so the problem now is clear 🙈 
add types to public properties leads to the problem:

`Typed property App\Http\Livewire\MInfo::$photo must not be accessed before initialization`

so what´s working here but makes things even more ugly:

```php
public TemporaryUploadedFile|string $photo;

public function __construct($id = null) {
        $this->photo = '';
        parent::__construct($id);
}
```
with that, the code should be type-safe. but I rather work without types than re-creating the constructor in every livewire-component.

if someone has a better idea, please let me know, thanks!

**Comment by joshhanley** [comment, marked as answer]
@michabbb ah yep, that's the hard thing with types for file uploads, as originally they are a string and then get converted to an the temp object.

I wouldn't use the constructor for that (as it's not considered part of the public API), just use the mount method

```php
public function mount()
{
    $this->photo = '';
}
```

Hope this helps!

**Reply by michabbb** [reply]
@joshhanley thanks, you´re right, no need for the constructor, putting this into "mount" is the best solution here and I can keep my types 😏 thanks for your help!

**Reply by joshhanley** [reply]
No worries 🙂
